conda_build/build.py

```diff
@@ -438,6 +454,99 @@ def create_info_files(m, files, config, prefix):
+            return target_file.replace("Scripts", "python-scripts")
+    elif m.get_value('build/noarch_python'):
+        return None
+    else:
+        return target_file
+
+
+def sha256_checksum(filename, buffersize=65536):
+    sha256 = hashlib.sha256()
+    with open(filename, 'rb') as f:
+        for block in iter(lambda: f.read(buffersize), b''):
+            sha256.update(block)
+    return sha256.hexdigest()
+
+
+def has_prefix(short_path, files_with_prefix):
+    for prefix, mode, filename in files_with_prefix:
+        if short_path == filename:
+            return prefix, mode
+    return None, None
+
+
+def is_no_link(no_link, short_path):
+    if no_link is not None and short_path in no_link:
+        return True
+
+
+def get_inode_paths(files, target_short_path, prefix):
+    ensure_list(files)
+    target_short_path_inode = os.stat(join(prefix, target_short_path)).st_ino
+    hardlinked_files = [sp for sp in files
+                        if os.stat(join(prefix, sp)).st_ino == target_short_path_inode]
+    return sorted(hardlinked_files)
+
+
+def file_type(path):
+    if isdir(path):
+        return FileType.directory
+    elif islink(path):
+        return FileType.softlink
+    return FileType.hardlink
+
+
+def build_info_files_json(m, prefix, files, files_with_prefix):
+    no_link = m.get_value('build/no_link')
+    files_json = []
+    for fi in files:
+        prefix_placeholder, file_mode = has_prefix(fi, files_with_prefix)
+        path = os.path.join(prefix, fi)
+        file_info = {
+            "short_path": get_short_path(m, fi),
+            "sha256": sha256_checksum(path),
+            "size_in_bytes": os.path.getsize(path),
+            "file_type": getattr(file_type(path), "name"),
+        }
+        no_link = is_no_link(no_link, fi)
+        if no_link:
+            file_info["no_link"] = no_link
+        if prefix_placeholder and file_mode:
+            file_info["prefix_placeholder"] = prefix_placeholder
+            file_info["file_mode"] = file_mode
+        if file_info.get("file_type") == "hardlink" and os.stat(join(prefix, fi)).st_nlink > 1:
+            inode_paths = get_inode_paths(files, fi, prefix)
+            file_info["inode_paths"] = inode_paths
+        files_json.append(file_info)
+    return files_json
+
+
+def get_files_version():
+    return 1
+
+
+def create_info_files_json(m, info_dir, prefix, files, files_with_prefix):
+    files_json_fields = ["short_path", "sha256", "size_in_bytes", "file_type", "file_mode",
+                         "prefix_placeholder", "no_link", "inode_first_path"]
+    files_json_files = build_info_files_json(m, prefix, files, files_with_prefix)
+    files_json_info = {
+        "version": get_files_version(),
+        "fields": files_json_fields,
+        "files": files_json_files,
+    }
+    with open(join(info_dir, 'files.json'), "w") as files_json:
+        json.dump(files_json_info, files_json)
```

Comment:
To dump the enums correctly, we might have to use something like `conda._vendor.auxlib.entity.EntityEncoder`.  Make *sure* `sort_keys=True` in the dump.  Might also want to add `indent=2, separators=(',', ': '),` to pretty it up a bit.